api/client/client.go

```diff
@@ -226,10 +226,10 @@ func (c *Client) DownloadDirectory(hash, path, destDir, credentials, publisher s
 	}
 
 	uri := c.Gateway + "/bzz:/" + hash + "/" + path
-	if publisher != "" {
-		uri += "?publisher=" + publisher
-	}
 	req, err := http.NewRequest("GET", uri, nil)
+	values := req.URL.Query()
```

Comment:
why not just 

     req.URL.Query().Add("publisher", publisher)

Reply:
I used `req.URL.Query().Add("publisher", publisher)` and the query param was not reaching the other side.
I looked at other places of the code using Add/Set and it was encoding after adding it.

After changing it to .Encode() it works correctly.
@mortelli saw the same [thing](https://github.com/ethersphere/swarm/pull/2202#discussion_r440843508)